common/test/java/com/couchbase/lite/DbCollectionsTest.kt

```diff
@@ -47,22 +48,71 @@ class DbCollectionsTest : BaseCollectionTest() {
         assertNotNull(scope.getCollection("Chintz"))
         assertNotNull(scope.getCollection("6hintz"))
         assertNotNull(scope.getCollection("-Ch1ntz"))
+
+        // collections exists when calling from database
+        assertNotNull(baseTestDb.getCollection("chintz"))
+        assertNotNull(baseTestDb.getCollection("Chintz"))
+        assertNotNull(baseTestDb.getCollection("6hintz"))
+        assertNotNull(baseTestDb.getCollection("-Ch1ntz"))
     }
 
 
     @Test(expected = CouchbaseLiteException::class)
-    fun testCollectionNameStartWithIllegalChars1() {
+    fun testCollectionNameStartsWithIllegalChars1() {
         baseTestDb.createCollection("_notvalid")
     }
 
     @Test(expected = CouchbaseLiteException::class)
-    fun tesCollectionNameStartWithIllegalChars2() {
+    fun testCollectionNameStartsWithIllegalChars2() {
         baseTestDb.createCollection("%notvalid")
     }
 
-    @Test(expected = CouchbaseLiteException::class)
+    @Test
     fun testCollectionNameContainingIllegalChars() {
-        baseTestDb.createCollection("notval!d")
+        val array = charArrayOf(
+            '!',
+            '@',
+            '#',
+            '$',
+            '^',
+            '&',
+            '*',
+            '(',
+            ')',
+            '+',
+            '.',
+            '<',
+            '>',
+            '?',
+            '[',
+            ']',
+            '{',
+            '}',
+            '=',
+            '“',
+            '‘',
+            '|',
+            '\\',
+            '/',
+            '`',
+            '~'
+        )
+        for (c in array) {
+            val colName = "notval" + c + "d"
+            try {
+                baseTestDb.createCollection(colName)
```

Comment:
Would this not be simpler:
try {
    baseTestDb.createCollection(colName)
    fail("Expect CBL Exception for collection : $colName")
} catch (e: CouchbaseLiteException) {  }